e2e/page_objects/programCatalog.js

```diff
@@ -30,21 +37,79 @@ class ProgramCatalog {
     }
 
     countCourses() {
-        return this.card.count();
+        return this.cards.count();
     }
 
     getPlaceholderText() {
         return this.searchBar.getAttribute('placeholder');
     }
 
     typeIntoSearchBar(text) {
-        return this.searchBar.sendKeys(text);
+        this.searchBar.sendKeys(text);
+        return browser.sleep(3000);
     }
 
     clearSearchBar() {
-        this.searchBar.sendKeys(protractor.Key.chord(protractor.Key.CONTROL,"a"));
+        this.searchBar.sendKeys(protractor.Key.chord(protractor.Key.CONTROL, "a"));
         return this.searchBar.sendKeys(protractor.Key.BACK_SPACE);
     }
+
+    selectedFilters() {
+        return this.selectedFiltersField.getText();
+    }
+
+    getResultCounter() {
+        return this.resultCounter.getText().then(text => {
+            let regex = /Results? \((\d+)\)/;
+            let token = text.match(regex);
+            return +token[1];
+        });
+    }
+
+    openFilterDropdown(text) {
+        return this.dropdownSelectorisVisible.isVisible().then(value => {
+            if (value === false){
+                return this.dropdownSelector(text).click();
+            }
+            else {
+                return false;
+            }
+        });
+    }
+
+    clickOnFilter(text) {
+        return this.dropdownFilterSelector(text).click();
+    }
+
+    areCourseLevelLogosCorrect(level) {
+        return this.cards.then(cards => {
+            return Promise.all(cards.map(card => {
+                return this.courseLevelLogo(card).getAttribute('class');
+            })).then(results => {
+                return results.every(result => {
+                    return result.indexOf(level) > -1;
+                });
+            });
+        });
+    }
+
+    correctCourseLevelTextIsVisible(level) {
+        return this.cards.then(cards => {
+            return Promise.all(cards.map(card => {
+                return this.courseLevelText(card).getText();
+            })).then(results => {
+                return results.every(result => {
+                    let isEqual = result.localeCompare(level);
```

Comment:
If you want to check if the result is the same as the level, you can use result === level.